src/components/GroupAll/GroupRecommend.tsx

```diff
@@ -1,46 +1,38 @@
-import { useEffect, useState } from 'react';
 import styled from 'styled-components';
 import { IcBtnInformation } from '../../assets';
 import RecommendCard from '../../common/RecommendCard';
-import { getRoomRecommend } from '../../libs/apis/GroupAll/getRoomRecommend';
+import useGetRoomRecommend from '../../libs/hooks/GroupAll/useGetRoomRecommend';
 
 const GroupRecommend = () => {
   const nickname = sessionStorage.getItem('nickname');
-  const [groupData, setGroupData] = useState([]);
 
-  useEffect(() => {
-    const fetchData = async () => {
-      const response = await getRoomRecommend();
+  const id = sessionStorage.getItem('user')!!;
+  const userId = parseInt(id);
 
-      if (response.code === 200) {
-        const data = response.data;
+  const { data, isLoading } = useGetRoomRecommend(userId);
 
-        if (data && data.rooms) {
-          const limitedRooms = data.rooms.slice(0, 6);
-          setGroupData(limitedRooms);
-        }
-      }
-    };
-
-    fetchData();
-  }, []);
+  const group = data?.data?.rooms?.slice(0, 6) || [];
```

Comment:
ASK: 여기 모든 부분을 nullable로 처리하신 이유가 궁금해요! 위의 코드에서 userId가 무조건 들어오는 경우로 단언을 해주었기 때문에, userId가 들어오지 않는 경우는 없고 서버에 해당 id로 그룹을 조회했을 때 보여줄 그룹이 없다면 어차피 빈 배열을 보내주지 않나요!?

Reply:
필요할 때만 옵셔널 체이닝을 사용해야 하는데...이 부분 수정하겠습니다!!
그리고 그룹이 없다면 빈 배열을 보여주니깐 따로 처리하지 않아도 될까요?
```
const group = data?.data.rooms.slice(0, 6);
```

Reply:
넵 그냥 둬도 괜찮을 것 같아요! 그런데 data에 옵셔널 체이닝을 걸어주지 않으면 에러가 발생하나용?!

Reply:
P2: 이렇게 간단하게 나타낼 수도 있어요!

```suggestion
  const { rooms } = !isLoading && data.data;
  const group = rooms.slice(0, 6);
```